src/psyclone/psyir/backend/fortran.py

```diff
@@ -1055,16 +1055,18 @@ def filecontainer_node(self, node):
         :rtype: str
 
         :raises VisitorError: if the attached symbol table contains \
-            any data symbols.
+            any non-routine symbols.
         :raises VisitorError: if more than one child is a Routine Node \
             with is_program set to True.
 
         '''
         if node.symbol_table.symbols:
-            raise VisitorError(
-                f"In the Fortran backend, a file container should not have "
-                f"any symbols associated with it, but found "
-                f"{len(node.symbol_table.symbols)}.")
+            for symbol in node.symbol_table.symbols:
+                if not isinstance(symbol, RoutineSymbol):
```

Comment:
Can you add a TODO here then (probably to the same issue 2201 since this a towards PR), saying that ContainerSymbols should also be accepted but currently this are stored to its containing scope.

Reply:
Added